styx-scheduler-service/src/main/java/com/spotify/styx/docker/KubernetesDockerRunner.java

```diff
@@ -285,7 +288,24 @@ private static EnvVar envVar(String name, String value) {
 
   @Override
   public void cleanup(String executionId) {
-    client.pods().withName(executionId).delete();
+    final Pod pod = client.pods().withName(executionId).get();
+    if (pod != null) {
+      cleanup(pod);
```

Comment:
This feels a bit overhead. The pod survived readPodWorkflowInstance check already when processing event.